app/controllers/concerns/sufia/files_controller_behavior.rb

```diff
@@ -268,5 +274,13 @@ def update_metadata_from_upload_screen
         @generic_file.relative_path = params[:relative_path] if params[:relative_path]
         @generic_file.on_behalf_of = params[:on_behalf_of] if params[:on_behalf_of]
       end
+
+      def add_file_to_collection
+        collection_id = params[:collection]
+        return if collection_id.nil? || collection_id == -1
+        collection = Collection.find(collection_id)
+        collection.member_ids = [@generic_file.id].concat(collection.member_ids)
+        ActiveFedora::SolrService.instance.conn.commit if collection.save
```

Comment:
.save should commit too... in fact, I'm pretty sure we've got autocommit enabled.